src/app/shared/file-editing.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ *    Copyright 2017 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import { Input } from '@angular/core';
+import { Files } from './files';
+import { SourceFile } from './swagger/model/sourceFile';
+
+export class FileEditing extends Files {
+  /**
+   * Toggles edit mode
+   * @return
+   */
+  toggleEdit() {
+    this.editing = !this.editing;
+  }
+
+  /**
+   * Retrieves all dockerfiles from the list of sourcefiles
+   * @param  sourceFiles Array of sourcefiles
+   * @return {Array<SourceFile>}      Array of test parameter files
+   */
+  getDockerFile(sourceFiles: Array<SourceFile>): Array<SourceFile> {
+    return sourceFiles.filter(
+      sourcefile => sourcefile.type === 'DOCKERFILE');
```

Comment:
all these look like swagger models may have it somewhere

Reply:
For some reason all of the enums are missing the _ (like DOCKSTORE_WDL is DOCKSTOREWDL) so I can't use enums.